File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
  l, r, len = 0, 0, nums.length
  while r < len
    r += 1 while r < len && nums[r] == nums[l]
    nums[l+1] = nums[r] if r < len
    l += 1
  end
  l
end

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  const len = nums.length;
  let [l, r] = [0, 0];
  while (r < len) {
    while (nums[r] === nums[l] && r < len) r++;
    if (r < len) nums[l + 1] = nums[r];
    l++;
  }
  return l;
};